src/common/shares.rs

```diff
@@ -1,24 +1,223 @@
 /// This file contains the more common secret sharing protocols used in MPC.
 /// You can reuse them for the MPC protocols that you aim to implement.
-/// 
+///
+use crate::{SecretSharing, Share, ShareError};
+use ark_ff::{FftField, Zero};
+use ark_poly::{univariate::DensePolynomial, DenseUVPolynomial, Polynomial};
+use ark_std::rand::Rng;
 
-use ark_ff::{FiniteField};
+// struct represents a single shamir share
+#[derive(Debug, Clone, Copy)]
+pub struct ShamirSecretSharing<F: FftField> {
+    pub share: F,      // shamir share
+    pub id: usize,     // id of the share
+    pub degree: usize, // degree of the underlying polynomial
+}
+
+impl<F: FftField> ShamirSecretSharing<F> {
+    pub fn new(share: F, id: usize, degree: usize) -> Self {
+        Self { share, id, degree }
+    }
+}
+
+impl<F: FftField> SecretSharing<ShamirSecretSharing<F>> for ShamirSecretSharing<F> {
+    // compute the shamir shares of all ids for a secret
+    fn compute_shares(
+        secret: <ShamirSecretSharing<F> as Share>::UnderlyingSecret,
```

Comment:
If the suggestion from the `lib.rs` file is taken, then this is simplified quite a bit and doesn't look as clunky.